### ISSUE TITLE
Windows build do not define O_NONBLOCK

### ISSUE THREAD

**Issue by pasha1st** [php/pecl-system-dio#9]
Windows version do not define O_NONBLOCK constant

**Comment by cmb69** [comment]
That is correct, because the C macro `O_NONBLOCK` is not defined on Windows.

**Comment by remicollet** [comment]
PR welcome,
I don't even know what Windows is ;)

**Comment by cmb69** [comment]
> I need to read from serial port without blocking the script on windows - is there a way for this?

For serial ports, [that might be possible](https://github.com/php/pecl-system-dio/blob/3fd676e99eccb123b1f4b34c74576b3d1503342a/php7/dio_win32.c#L453-L478). Try something like:
````php
$com = dio_serial("COM1", "r");
stream_set_blocking($com, false);
````

**Comment by pasha1st** [comment]
$port='\\\\.\COM16';
$f=dio_open($port,O_RDWR);
var_dump(stream_set_blocking($f,false));

Warning: stream_set_blocking(): supplied resource is not a valid stream resource

**Comment by cmb69** [comment]
Try `dio_serial()` instead of `dio_open()`.

**Comment by pasha1st** [comment]
There is no description for "dio_serial".
I've got it requires 2 or 3 parameters - filename, mode, options (optional)
Every use of dio_serial generates PHP NOTICE
It is not clear what mode should I use for R/W: rw or r+
I can not use dio_serial('COM16','r'), I should use dio_serial('\\\\.\COM16','r')
I can not use this handle at all:
```
$port='\\\\.\COM16';
$f=dio_serial($port,'r');
$r=dio_read($f,4);
```
Warning: dio_read(): supplied resource is not a valid Direct I/O File Descriptor resource in

**Comment by cmb69** [comment]
Try `fread($f, 4)` instead of `dio_read($f, 4)`. And if you need R/W, try `dio_serial($port,'r+');`.

Sorry, I cannot check that myself because I don't have any COM ports on my machine (not sure why).

**Comment by pasha1st** [comment]
Yes, this code works:
```
$port='\\\\.\COM16';
$f=@dio_serial($port,'r+');
var_dump(stream_set_blocking($f,false));
$r=fwrite($f,"ATE0\r\n");
usleep(1000000);
do {
  $r=fread($f,1024);
  print $r;
} while ($r!='');

```

**Comment by cmb69** [comment]
> Yes, this code works:

Great!

So this is basically a documentation issue.
